music21/note.py

```diff
@@ -1190,6 +1192,22 @@ def _setVolume(self, value, setClient=True):
         [120, 80]
         ''')
 
+    def _getStoredInstrument(self):
+        return self._storedInstrument
+
+    def _setStoredInstrument(self, newValue):
+        self._storedInstrument = newValue
+
+    storedInstrument = property(_getStoredInstrument, _setStoredInstrument)
+
+    def getInstrument(self):
+        '''
+        TOOD: doc and test
+        '''
+        if self.storedInstrument is not None:
+            return self.storedInstrument
+        return self.getContextByClass('Instrument', followDerivation=False)
```

Comment:
not sure whether followDerivation should be True or False here.

Reply:
what are the most likely derivations? stripTies? partitionByInstrument? I think those keep instruments around, so I think we're alright. We just added instruments to voicesToParts.

Reply:
I'm thinking it should be False. I just went on a goose chase in #1051 dealing with instruments being found that weren't in the stream being exported and getting mismatches with the `<part-list>`.

Reply:
In other words, it's not really "stored" if the instrument is not in the stream. So, maybe: If you want derivations, then opt-in with `gCBC(follow...`)